### PR TITLE
Allow users to add external accounts

### LINKED ISSUE #2480: Allow users to add external accounts to their profile
Reference: [specs](https://github.com/mozilla/pontoon/blob/master/specs/0111-redesign-profile-page.md)

Users should be able to provide a series of external accounts to help locale managers and administrators contacting them:
* Bugzilla (email address).
* GitHub (username).
* Matrix (username).

### PR DESCRIPTION
**Pull request by mathjazz**
Fixes #2480.

For now the data is not used anywhere. It will be presented on the new [profile page](https://github.com/mozilla/pontoon/projects/4).

### CODE REVIEW

File: pontoon/base/migrations/0029_external_accounts.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        ("base", "0028_userprofile_new_contributor_notifications"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="userprofile",
+            name="bugzilla",
+            field=models.EmailField(
+                blank=True,
+                max_length=254,
```
Comment (eemeli) [inline pontoon/base/migrations/0029_external_accounts.py:18]:
```suggestion
                max_length=255,
```

254 is a bit of a weird length, any particular reason?

File: pontoon/base/models.py [hunk 2]
```diff
@@ -1591,6 +1591,10 @@ class UserProfile(models.Model):
     # Used to display strings from preferred source locale.
     preferred_source_locale = models.CharField(max_length=20, blank=True, null=True)
 
+    bugzilla = models.EmailField("Bugzilla email address", blank=True, null=True)
```
Comment (eemeli) [inline pontoon/base/models.py:1594]:
```suggestion
    bugzilla = models.EmailField("Bugzilla email address", max_length=255, blank=True, null=True)
```

**Review by eemeli — APPROVED** [review]
Looks good. A few ignorable nitpicks below.

File: pontoon/base/migrations/0029_external_accounts.py (hunk 1 above)
Comment (mathjazz) [reply pontoon/base/migrations/0029_external_accounts.py:18]: That is auto generated and implicitly set by EmailField:
https://docs.djangoproject.com/en/4.0/ref/models/fields/#emailfield

Here's the background:
https://www.rfc-editor.org/errata_search.php?rfc=3696&eid=1690

**Pull request merged**